changes for combined append+flush calls for blockblob(traditionan) an…

ishaniahuja opened the pull request:
…d support for appendblob

Commit:
changes for combined append+flush calls for blockblob(traditionan) and support for appendblob

bilaharith commented on hadoop-tools/hadoop-azure/pom.xml:
```diff
@@ -143,6 +143,14 @@
+		<dependency>
+				<groupId>org.mockito</groupId>
+				<artifactId>mockito-all</artifactId>
+				<version>1.8.4</version>
```
The practice I observed is that we are specifying the version in the parent pom

bilaharith commented on hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java:
```diff
@@ -164,6 +169,23 @@ public AzureBlobFileSystemStore(URI uri, boolean isSecureScheme, Configuration c
     initializeClient(uri, fileSystemName, accountName, useHttps);
     this.identityTransformer = new IdentityTransformer(abfsConfiguration.getRawConfiguration());
+
+    // Extract the directories that should contain page blobs
```
This constructor is becoming lengthy. I would recommend make this code addition as a separate private method and call it from this constructor, so that the constructor would look clean.

snvijaya commented on hadoop-tools/hadoop-azure/pom.xml:
```diff
@@ -143,6 +143,14 @@
 
```
Please change the merge branch to apache/hadoop - trunk.

Commit:
resolving conflicts
+ some changes were not checked in with the previous commit and are checked in here.

Commit:
added feedbacks

Commit:
adding fixes

Commit:
made rest version based on config parameter

Commit:
incorporated feedbacks/comments

Commit:
added appropriate comments

Commit:
compilation error fixes

Commit:
fixed UTs

Commit:
fixing mockito errors

Commit:
adding right version of mockito

Commit:
fixed errors

Commit:
fixes in AbfsClient.java

Commit:
modification to test due to the new call changes in the driver

Commit:
whitespace fix

Commit:
whitespace fix

Commit:
fixes for appendblob changes, + added comment to change in the test cases. tested with both namespace enabled and disabled accounts

Commit:
incorporated feedbacks/comments

Commit:
incoroporated feedbacks in the TestAbfsOutputStream

Commit:
removed mockito version from the pom file

Commit:
incorporated feedbacks

Commit:
incorporated feedbacks

Commit:
incorporated feedbacks
